### DISCUSSION TITLE
Deploy Button support for Turborepo

### DISCUSSION THREAD

**Discussion by peterfritz** [vercel/vercel#7725] [Ideas] [answered]
Turborepo [requires](https://vercel.com/docs/concepts/git/monorepos#turborepo) you to set the build command and the root directory on Vercel as you deploy the project, but on the [Deploy Button documentation](https://vercel.com/docs/deploy-button), there's no instruction on how to set these using URL params.

Is it malpractice to manage a open-source software within a monorepo with a deploy button for only one of the apps?

Cc. @lfades (Creator of issue [#719](https://github.com/vercel/turborepo/issues/719) on Turborepo's repository)

**Comment by lfades** [comment, marked as answer]
> Is it malpractice to manage a open-source software within a monorepo with a deploy button for only one of the apps?

Can you provide more context on this? from my understanding it's not a bad practice to do that.

**Reply by peterfritz** [reply]
I don't think it is either. I was just curious as to why that's not a feature yet. I thought it was in favor of security, performance, or so Vercel doesn't have to download and store the codebase of five apps only to run one.

I actually think having the documentation, mobile, and web apps in the same repo greatly improves the project's maintainability and contribution process as a PR could update everything related to the changes it makes.

**Reply by lfades** [reply]
I agree and I think we might take a while to decide what to do here, there are indeed some things we are not 100% sure about yet about using query params to add those settings, it was added to add quick support for monorepos in the deploy button flow as it wasn't possible before. It's probably better to add it to the Vercel deploy flow UI or improve the way we set the settings for monorepos before adding the docs.

**Reply by peterfritz** [reply]
So there is a way to pass on those parameters using the Deploy Button's flow?
I thought the logic had yet to be implemented.

**Reply by lfades** [reply]
@peterfritz yes, it's possible. The issue I added is about adding those fields to the deploy button as they already work, but aren't documented.

**Reply by peterfritz** [reply]
I also think the feature should be documented, I've been searching for a way to do that for days!
Could you please link a resource or send me the related query parameters?

**Reply by lfades** [reply]
It's the same query parameters mentioned in the issue, you can also see them in use in the deploy button here: https://github.com/vercel/examples/tree/main/solutions/monorepo

**Reply by peterfritz** [reply]
Oh my god, I'm so sorry.
Thank you so much, you're awesome.
